CHANGES.md

```diff
@@ -0,0 +1,18320 @@
+   and L<EVP_PKEY_derive(3)>.
+
+   *Paul Dale*
+
+ * All of the low level DSA functions have been deprecated including:
+
+   DSA_do_sign, DSA_do_verify, DSA_OpenSSL, DSA_set_default_method,
+   DSA_get_default_method, DSA_set_method, DSA_get_method, DSA_new_method,
+   DSA_sign_setup, DSA_sign, DSA_verify, DSA_get_ex_new_index,
+   DSA_set_ex_data, DSA_get_ex_data, DSA_generate_parameters_ex,
+   DSA_generate_key, DSA_meth_new, DSA_get0_engine, DSA_meth_free,
+   DSA_meth_dup, DSA_meth_get0_name, DSA_meth_set1_name, DSA_meth_get_flags,
+   DSA_meth_set_flags, DSA_meth_get0_app_data, DSA_meth_set0_app_data,
+   DSA_meth_get_sign, DSA_meth_set_sign, DSA_meth_get_sign_setup,
+   DSA_meth_set_sign_setup, DSA_meth_get_verify, DSA_meth_set_verify,
+   DSA_meth_get_mod_exp, DSA_meth_set_mod_exp, DSA_meth_get_bn_mod_exp,
+   DSA_meth_set_bn_mod_exp, DSA_meth_get_init, DSA_meth_set_init,
+   DSA_meth_get_finish, DSA_meth_set_finish, DSA_meth_get_paramgen,
+   DSA_meth_set_paramgen, DSA_meth_get_keygen and DSA_meth_set_keygen.
+
+   Use of these low level functions has been informally discouraged for a long
+   time.  Instead applications should use L<EVP_DigestSignInit_ex(3)>,
+   L<EVP_DigestSignUpdate(3)> and L<EVP_DigestSignFinal(3)>.
+
+   *Paul Dale*
+
+ * Reworked the treatment of EC EVP_PKEYs with the SM2 curve to
+   automatically become EVP_PKEY_SM2 rather than EVP_PKEY_EC.
+   This means that applications don't have to look at the curve NID and
+   `EVP_PKEY_set_alias_type(pkey, EVP_PKEY_SM2)` to get SM2 computations.
+   However, they still can, that EVP_PKEY_set_alias_type() call acts as
+   a no-op when the EVP_PKEY is already of the given type.
+
+   Parameter and key generation is also reworked to make it possible
+   to generate EVP_PKEY_SM2 parameters and keys without having to go
+   through EVP_PKEY_EC generation and then change the EVP_PKEY type.
+   However, code that does the latter will still work as before.
+
+   *Richard Levitte*
+
+ * Deprecated low level ECDH and ECDSA functions.  These include:
+
+   ECDH_compute_key, ECDSA_do_sign, ECDSA_do_sign_ex, ECDSA_do_verify,
+   ECDSA_sign_setup, ECDSA_sign, ECDSA_sign_ex, ECDSA_verify and
+   ECDSA_size.
+
+   Use of these low level functions has been informally discouraged for a long
+   time.  Instead applications should use the EVP_PKEY_derive(3),
+   EVP_DigestSign(3) and EVP_DigestVerify(3) functions.
+
+   *Paul Dale*
+
+ * Deprecated the EC_KEY_METHOD functions.  These include:
+
+   EC_KEY_METHOD_new, EC_KEY_METHOD_free, EC_KEY_METHOD_set_init,
+   EC_KEY_METHOD_set_keygen, EC_KEY_METHOD_set_compute_key,
+   EC_KEY_METHOD_set_sign, EC_KEY_METHOD_set_verify,
+   EC_KEY_METHOD_get_init, EC_KEY_METHOD_get_keygen,
+   EC_KEY_METHOD_get_compute_key, EC_KEY_METHOD_get_sign and
+   EC_KEY_METHOD_get_verify.
+
+   Instead applications and extension writers should use the OSSL_PROVIDER APIs.
+
+   *Paul Dale*
+
+ Deprecated EVP_PKEY_decrypt_old(), please use EVP_PKEY_decrypt_init()
```

Comment:
Missing `*` (found visually by having a look at https://github.com/mspncp/openssl/blob/revamp-the-docs/CHANGES.md).

Seems to be the only case since 1.1.1; I didn't check further back.

Reply:
Thanks, fixed by 784f17aa2b37073637d3d504e97d634ad327bc4e.